### PR TITLE
fix: change default gesture from Tap to Fling, fix failing CI

### PR DESCRIPTION
**Pull request by tboba**
## Description

Unfortunately, the `Tap` gesture as the default gesture in ScreenGestureDetector breaks the back button on Android and iOS. Somehow, gesture detector while "tapping" just taps and holds the back button, resulting in breaking CI and our testing apps.
This PR changes this gesture to `Fling`, resulting in correct behavior. It cannot be changed to `Manual` gesture, since it also breaks the iOS CI.

## Changes

- Changed gesture from Tap to Fling
- Updated detox from 20.11.4 to 20.17.0
- Fixed Android E2E tests by building the bundle with .so files on x86_64 architecture
- Changed scroll action that was opening background apps preview on Android 33 (and higher) levels
- Adjusted CI parameters, fixed the typo of artifact names

## Test code and steps to reproduce

You can go through Example app and try to click back button in `Events`. Right now it should work properly.

## Checklist

- [x] Ensured that CI passes

### CODE REVIEW

File: Example/e2e/examplesTests/events.e2e.ts [hunk 1]
```diff
@@ -8,7 +8,7 @@ describe('Events', () => {
     await waitFor(element(by.id('root-screen-playground-Events')))
       .toBeVisible()
       .whileElement(by.id('root-screen-examples-scrollview'))
-      .scroll(200, 'down');
+      .scroll(100, 'down', NaN, 0.85);
```
Comment (WoLewicki) [inline Example/e2e/examplesTests/events.e2e.ts:11]: What is the reason for changing this value?

**Comment by WoLewicki** [comment]
Also, I am not sure I understand what is going on exactly. Why does gesture-handler call the `tap`? Are we testing custom animations on CI?

**Comment by tboba** [comment]
@WoLewicki we're not testing them, but this change does not only limit to the case of the CI. Because of the default `Tap` option as an empty gesture, Android and iOS simulate that motion as holding the element (I don't specifically know why), resulting in unability to click "go back" button, because it detects the click as the hold motion.

**Comment by WoLewicki** [comment]
Ok keep me updated 🚀

**Comment by tboba** [comment]
@WoLewicki @kkafar I think the changes are ready! 🚀 
Could you take a look on them?

**Review by kkafar — COMMENTED** [review]
I don't quite get why our tests start to fail. Is it because in `Example` app we now wrap whole application view hierarchy in `GestureDetectorProvider` and gestures are hijacked? Wouldn't it be better to detect from environment whether we are in e2etest scenario and just don't wrap it?

**Review by WoLewicki — APPROVED** [review]
I don't like the idea that we are changing the library code only for the e2e tests to pass, but if we are 100% sure it works correctly then it is ok to me.

**Pull request merged**